### PR TITLE
Add achievements api

### PR DESCRIPTION
**Pull request by ssha0054**
# Description 
Added api requests to create a new achievement, get all achievements for a team, update an achievement and delete an achievement.

# Type of change 
- [ ] Bug fix
- [x] New feature
- [ ] Refactoring
- [ ] Documentation 

# Problem
Api endpoints are needed for the awards and achievement page.

# Solution description
- Create a new achievement via a `POST` request on the endpoint `/achievements/`
- Get all achievements via a `GET` request on the endpoint `/achievements/team/:team_id` , where `:team_id` is the mongo object id for the team.
- Delete an achievement via a `DELETE` request on the endpoint `/achievements/:id` , where `:id` is the mongo object id for the achievement
- Update an achievement via a `PATCH` request on the endpoint `/achievements/:id`, where `:id` is the mongo object id for the achievement

# Tests 
Tested the api requests on Insomnia. 

All unit tests pass? Yes

Builds successfully? Yes

# Relevant document/ link (if any) 
N/A

# Risk
Identity the risk level (low, medium, high) and indicate which component(s) will be affected if bugs exist.
Low

### CODE REVIEW

**Review by pmyip3 — CHANGES_REQUESTED** [review]
Can you use async/ await instead of promises? 
As we've created a ticket on converting all the existing controllers from promises to async/ await based on the previous discussion https://github.com/Researchify/Researchify/pull/140#discussion_r681393574
Would be nice if the incoming tickets are coded in async/ await style.

**Comment by ssha0054** [comment]
> Can you use async/ await instead of promises?
> As we've created a ticket on converting all the existing controllers from promises to async/ await based on the previous discussion [#140 (comment)](https://github.com/Researchify/Researchify/pull/140#discussion_r681393574)
> Would be nice if the incoming tickets are coded in async/ await style.

Yeah no worries, will make the changes.

**Review by rpra0006 — APPROVED** [review]

File: api/src/controllers/achievements.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+ * @param res response object - a list of achievements (see Achievement model)
+ * @returns 200: a list of achievements by the given team id
+ * @returns 500: server encountered an unexpected condition
+ */
+async function getAllAchievementsByTeam(req, res) {
+    const { team_id: _id } = req.params;
+    
+    try {
+        // Store achievements into list, sorted by title in alphabetical order
+        const foundAchievement = await Achievement.aggregate([
```
Comment (aksh0001) [inline api/src/controllers/achievements.js:53]: nit: make this name plural.

**Review by aksh0001 — APPROVED** [review]
please see nit comments, but otherwise LGTM

**Review by pmyip3 — APPROVED** [review]

**Comment by vgoh1** [comment]
@ssha0054 Remember to hit squash and merge so the commits don't clog up the main branch, and also remember to delete your branch next time! Thanks!

**Pull request merged**